File: 1701-average-waiting-time/1701-average-waiting-time.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        double sum = 0.0;
        int end = customers[0][0];
        for(vector<int> customer: customers) {
            end = max(end + customer[1], customer[0] + customer[1]);
            sum += end - customer[0];
        }
        return sum / customers.size();
    }
};

File: 1701-average-waiting-time/1701-average-waiting-time.go
func averageWaitingTime(customers [][]int) float64 {
    var sum int = 0;
    var end int = customers[0][0];
    for _, customer := range customers {
        end = int(math.Max(float64(end + customer[1]), float64(customer[0] + customer[1])));
        sum += end - customer[0];
    }
    return float64(sum) / float64(len(customers));
}


